File: Patient_Client/WebContent/Components/Patient.js
$(document).ready(function() 
{
	if ($("#alertSuccess").text().trim() == "")
	{
		$("#alertSuccess").hide();
	}
	$("#alertError").hide();
});

// SAVE ============================================
$(document).on("click", "#btnSave", function(event) {

	// Clear status msges-------------
	$("#alertSuccess").text("");
	$("#alertSuccess").hide();
	$("#alertError").text("");
	$("#alertError").hide();

	// Form validation----------------
	var status = validatePatientForm();
	if (status != true) 
	{
		$("#alertError").text(status);
		$("#alertError").show();
		return;
	}
	
	// If valid-----------------------
	var type = ($("#hidItemIDSave").val() == "") ? "POST" : "PUT";
	$.ajax( 
			{
				url : "PatientAPI",  
				type : type,  data : 
					$("#formPatient").serialize(), 
					dataType : "text",  
					complete : function(response, status)  
					{   onPatientSaveComplete(response.responseText, status); 
			} 
		}); 
});
function onPatientSaveComplete(response, status) {
	
	var resultSet = JSON.parse(response); 
	 
	if (resultSet.status.trim() == "success") 
	{  
			$("#alertSuccess").text("Successfully saved."); 
			$("#alertSuccess").show(); 
			$("#divItemsGrid").html(resultSet.data); 
			} 
	else if (resultSet.status.trim() == "error")
	{ 
		$("#alertError").text(resultSet.data); 
		$("#alertError").show(); }

	 else if (status == "error") 
	 { 
		 $("#alertError").text("Error while saving."); 
		 $("#alertError").show();
	 }
	 else 
	 { 
		 $("#alertError").text("Unknown error while saving.."); 
		 $("#alertError").show(); 
		 $("#hidItemIDSave").val(""); 
		 $("#formItem")[0].reset(); 
	 }
}
// UPDATE=====================================================================
$(document).on("click",".btnUpdate",function(event) {
			$("#hidItemIDSave").val($(this).closest("tr").find('#hidItemIDUpdate').val());
			$("#FirstName").val($(this).closest("tr").find('td:eq(0)').text());
			$("#LastName").val($(this).closest("tr").find('td:eq(1)').text());
			$("#NIC").val($(this).closest("tr").find('td:eq(2)').text());
			$("#DOB").val($(this).closest("tr").find('td:eq(3)').text());
			$("#Email").val($(this).closest("tr").find('td:eq(4)').text());
			$("#Mobile").val($(this).closest("tr").find('td:eq(5)').text());
			$("#Address").val($(this).closest("tr").find('td:eq(6)').text());
			$("#BloodGroup").val($(this).closest("tr").find('td:eq(7)').text());
			$("#Allergy").val($(this).closest("tr").find('td:eq(8)').text());
			$("#Gender").val($(this).closest("tr").find('td:eq(9)').text());
			$("#Password").val($(this).closest("tr").find('td:eq(10)').text());
		});

//REMOVE========================================================================
$(document).on("click", ".btnRemove", function(event) 
		{  
	$.ajax( 
			{   
				url : "PatientAPI",  
				type : "DELETE",   
				data : "Patient_ID=" + $(this).data("Patient_ID"),   
				dataType : "text",   
				complete : function(response, status)  
				{    
					onPatientSaveComplete(response.responseText, status);  
					} 
			}); 
	}); 
function onPatientSaveComplete(response, status)
{ 
	if (status == "success")  
	{  
		var resultSet = JSON.parse(response); 

if (resultSet.status.trim() == "success")  
{   
	$("#alertSuccess").text("Successfully deleted.");  
	$("#alertSuccess").show(); 
    $("#divItemsGrid").html(resultSet.data);   
 } 
else if (resultSet.status.trim() == "error")  
{   
	$("#alertError").text(resultSet.data);   
	$("#alertError").show();   
	} 

} else if (status == "error")  
{  
	$("#alertError").text("Error while deleting."); 
	$("#alertError").show();  } 
else  
{ 
		$("#alertError").text("Unknown error while deleting.."); 
		$("#alertError").show();  
		} 
	}

// CLIENT-MODEL=================================================================
function validatePatientForm() {
	// FirstName
	if ($("#FirstName").val().trim() == "") {
		return "Insert Patient First Name.";
	}

	// LastName
	if ($("#LastName").val().trim() == "") {
		return "Insert Patient Last Name.";
	}
	// NIC-------------------------------
	if ($("#NIC").val().trim() == "") {
		return "Insert NIC.";
	}
	// DOB-------------------------------
	if ($("#DOB").val().trim() == "") {
		return "Insert Date of Birth.";
	}
	// Email-------------------------------
	if ($("#Email").val().trim() == "") {
		return "Insert Email.";
	}
	// Mobile-------------------------------
	if ($("#Mobile").val().trim() == "") {
		return "Insert Mobile.";
	}
	// Address-------------------------------
	if ($("#Address").val().trim() == "") {
		return "Insert Address.";
	}
	// BloodGroup-------------------------------
	if ($("#BloodGroup").val().trim() == "") {
		return "Select BloodGroup.";
	}
	// Allergy-------------------------------
	if ($("#Allergy").val().trim() == "") {
		return "Insert Allergies.";
	}
	// Gender-------------------------------
	if ($("#Gender").val().trim() == "") {
		return "Select Gender.";
	}
	
	// Password------------------------
	if ($("#password").val().trim() == "") {
		return "Insert Patient Password.";
	}

	// ConfirmPassword------------------------
	if ($("#ConfirmPassword").val().trim() == "") {
		return "Confirm Patient Password.";
	}
	return true;
}

File: Patient_Client/src/com/Patient.java
package com;

import java.sql.*;

public class Patient {

public Connection connect() {
		Connection con = null;

		try {
			Class.forName("com.mysql.jdbc.Driver");
			con = DriverManager.getConnection("jdbc:mysql://127.0.0.1:3306/patient","root","");

			// For testing
			System.out.print("Successfully connected");
		} catch (Exception e) {
			e.printStackTrace();
		}
		return con;
	}
public String insertPatient(String FirstName, String LastName, String NIC,String DOB,String Email,String Mobile,String Address,String BloodGroup, String Allergy,String Gender,String Password,String ConfirmPassword)
		
		 
{
		String output = "";

		try {

			Connection con = connect();

			if (con == null) 
			{
				return "Error while connecting to the database for inserting.";
			}

			// create a prepared statement
			String query = " insert into patient(`Patient_ID`,`FirstName`,`LastName`,`NIC`,`DOB`,`Email`,`Mobile`,`Address`,`BloodGroup`,`Allergy`,`Gender`,`password`,`ConfirmPassword`)" + " values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)"; 
			 
			 
			
			PreparedStatement preparedStmt = con.prepareStatement(query);

			// binding values
			preparedStmt.setInt(1, 0);
			preparedStmt.setString(2, FirstName);
			preparedStmt.setString(3, LastName);
			preparedStmt.setString(4, NIC);
			preparedStmt.setString(5, DOB);
			preparedStmt.setString(6, Email);
			preparedStmt.setString(7, Mobile);
			preparedStmt.setString(8, Address);
			preparedStmt.setString(9, BloodGroup);
			preparedStmt.setString(10, Allergy);
			preparedStmt.setString(11, Gender);
			preparedStmt.setString(12, Password);
			preparedStmt.setString(13, ConfirmPassword);
		
			// execute the statement
			preparedStmt.execute();
			con.close();
			
			String newpatient = readPatient();
			output = "{\"status\":\"success\", \"data\": \"" + newpatient + "\"}";
			
		} catch (Exception e) {
			output = "{\"status\":\"error\", \"data\":         \"Error while inserting the Details.\"}";
			System.err.println(e.getMessage());
		}

		return output;
	}

public String readPatient()  
{   
	String output = ""; 

	try   
	{    
	Connection con = connect(); 
	if (con == null)   
	{return "Error while connecting to the database for reading."; } 
	 
	   // Prepare the html table to be displayed    
	output = "<table border=\'1'><tr><th>First Name</th><th>Last Name</th><th>NIC</th><th>DOB</th><th>Email</th><th>Mobile</th><th>Address</th><th>Blood Group</th><th>Allergy</th><th>Gender</th><th>password</th><th>ConfirmPassword</th><th>Update</th><th>Remove</th></tr>"; 
	

	 
	 
	   String query = "select * from patient";    
	   Statement stmt = con.createStatement();    
	   ResultSet rs = stmt.executeQuery(query); 
	   
	// iterate through the rows in the result set   
	   while (rs.next())   
	   {   
		    String Patient_ID = Integer.toString(rs.getInt("Patient_ID"));
		    String FirstName = rs.getString("FirstName");     
		    String LastName = rs.getString("LastName");     
		    String NIC = rs.getString("NIC"); 
		    String DOB = rs.getString("DOB"); 
		    String Email = rs.getString("Email"); 
		    String Mobile = rs.getString("Mobile"); 
		    String Address = rs.getString("Address"); 
		    String BloodGroup = rs.getString("BloodGroup"); 
		    String Allergy = rs.getString("Allergy"); 
		    String Gender = rs.getString("Gender"); 
		    String Password = rs.getString("Password"); 
		    String ConfirmPassword = rs.getString("ConfirmPassword");
		    
		    // Add into the html table
			output += "<tr><td><input id='hidPatientUpdate'       name='hidPatientUpdate' type='hidden'        value='"
					+ Patient_ID + "'>" + FirstName + "</td>";     
		    output += "<td>" + LastName + "</td>";     
		    output += "<td>" + NIC + "</td>";     
		    output += "<td>" + DOB + "</td>"; 
		    output += "<td>" + Email + "</td>"; 
		    output += "<td>" + Mobile + "</td>"; 
		    output += "<td>" + Address + "</td>"; 
		    output += "<td>" + BloodGroup + "</td>"; 
		    output += "<td>" + Allergy + "</td>";
		    output += "<td>" + Gender + "</td>"; 
		    output += "<td>" + Password + "</td>"; 
		    output += "<td>" + ConfirmPassword + "</td>";
		 // buttons
			output += "<td><input name='btnUpdate' type='button'       value='Update'           class='btnUpdate btn btn-secondary'></td>      <td><input name='btnRemove' type='button'       value='Remove'           class='btnRemove btn btn-danger' data-Patient_ID='"
					+ Patient_ID + "'>" + "</td></tr>";
		    
	   } 
		    
		    con.close(); 
		  
		    // Complete the html table    
		    output += "</table>";   
		}   
		catch (Exception e)   
		{    
				output = "Error while reading the Patient Details.";    
				System.err.println(e.getMessage());   } 
	 
	  return output;  
	  
}
public String updatePatient(String Patient_ID,String FName, String LName, String NIC,String DOB,String Email,String Mobile,String Address,String BloodGroup, String Allergy,String Gender,String password,String ConfirmPassword) 
	{  
		String output = ""; 
	
	
		try   { 
				Connection con = connect(); 
		 
		   if (con == null)   
		   {return "Error while connecting to the database for updating."; } 
		 
		   // create a prepared statement   
		   String query = "UPDATE patient SET FirstName=?,LastName=?,NIC=?,DOB=?,Email=?,Mobile=?,Address=?,BloodGroup=?,Allergy=?,Gender=?,password=?,ConfirmPassword=? WHERE Patient_ID=?"; 
		 
		   PreparedStatement preparedStmt = con.prepareStatement(query);
		  
		   

		// binding values
			
					preparedStmt.setString(1, FName);
					preparedStmt.setString(2, LName);
					preparedStmt.setString(3, NIC);
					preparedStmt.setString(4, DOB);
					preparedStmt.setString(5, Email);
					preparedStmt.setString(6, Mobile);
					preparedStmt.setString(7, Address);
					preparedStmt.setString(8, BloodGroup);
					preparedStmt.setString(9, Allergy);
					preparedStmt.setString(10, Gender);
					preparedStmt.setString(11, password);
					preparedStmt.setString(12, ConfirmPassword);
					preparedStmt.setInt(13, Integer.parseInt(Patient_ID)); 
		    // execute the statement   
		   preparedStmt.execute();    
		   con.close(); 

			String newpatient = readPatient();
			output = "{\"status\":\"success\", \"data\": \"" + newpatient + "\"}";
		} catch (Exception e) {
			output = "{\"status\":\"error\", \"data\":         \"Error while updating the Details.\"}";
			System.err.println(e.getMessage());
		}

		return output;
	}
public String deletePatient(String Patient_ID) 
	{   
		String output = ""; 
	
		try   
		{   
			Connection con = connect(); 
	
			if (con == null)   
			{return "Error while connecting to the database for deleting."; } 
	
			// create a prepared statement    
			String query = "delete from patient where Patient_ID=?"; 
	
			PreparedStatement preparedStmt = con.prepareStatement(query); 
	
			// binding values    
			preparedStmt.setInt(1, Integer.parseInt(Patient_ID)); 
	
			// execute the statement    
			preparedStmt.execute();    
			con.close(); 
			
			String newpatient = readPatient();
			output = "{\"status\":\"success\", \"data\": \"" + newpatient + "\"}";
		} 
		catch (Exception e)
		{
			output = "{\"status\":\"error\", \"data\":         \"Error while deleting the Patient Details.\"}";
			System.err.println(e.getMessage());
		}

		return output;
	}
}